scripts/manage_translations.py

```diff
@@ -50,54 +55,79 @@ def list_resources_with_updates(date_since, date_skip=None, verbose=False):
-            last_update = lang_attributes["last_translation_update"]
-            if verbose:
-                print(
-                    f"CHECKING {resource_name} for {lang_id=} updated on {last_update}"
-                )
-            if last_update is None:
-                resource_lang_unchanged[resource_name].append(lang_id)
-                continue
+    return data
 
-            last_update = datetime.strptime(last_update, "%Y-%m-%dT%H:%M:%SZ")
-            if last_update > date_since and (
-                date_skip is None or last_update.date() != date_skip.date()
-            ):
-                if verbose:
-                    print(f"=> CHANGED {lang_attributes=} {date_skip=}")
-                resource_lang_changed[resource_name].append(lang_id)
-            else:
-                resource_lang_unchanged[resource_name].append(lang_id)
 
+def list_resources_with_updates(
+    date_since, resources=None, languages=None, verbose=False
+):
+    api_token = get_api_token()
+    project = "o:django:p:django"
+    date_since_iso = date_since.isoformat().strip("Z") + "Z"
     if verbose:
-        unchanged = "\n".join(
-            f"\n * resource {res} languages {' '.join(sorted(langs))}"
-            for res, langs in resource_lang_unchanged.items()
+        print(f"\n\n\n=== list_resources_with_updates, starting at {date_since_iso=}")
+
+    if not languages:
+        languages = [  # List languages using Transifex projects API.
+            d["attributes"]["code"]
+            for d in get_api_response(f"projects/{project}/languages", api_token)
+        ]
+    if verbose:
+        print(f"\n== List of languages: {languages=}")
+
+    if not resources:
+        resources = [  # List resources using Transifex resources API.
+            d["attributes"]["slug"]
+            for d in get_api_response(
+                "resources", api_token, params={"filter[project]": project}
+            )
+        ]
+    else:
+        resources = [_tx_resource_slug_for_name(r) for r in resources]
+    if verbose:
+        print(f"\n== List of resources: {resources=}")
+
+    resource_lang_changed = defaultdict(list)
+    for lang, resource in product(languages, resources):
+        if verbose:
+            print(f"\n=== Getting data for: {lang=} {resource=} {date_since_iso=}")
+        data = get_api_response(
+            "resource_translations",
+            api_token,
+            params={
+                "filter[resource]": f"{project}:r:{resource}",
+                "filter[language]": f"l:{lang}",
+                "filter[date_translated][gt]": date_since_iso,
+            },
         )
-        print(f"== SUMMARY for unchanged resources ==\n{unchanged}")
+        local_resource = resource.replace("contrib-", "", 1)
+        local_lang = lang  # XXX: LANG_OVERRIDES.get(lang, lang)
+        if data:
+            resource_lang_changed[local_resource].append(local_lang)
+            if verbose:
```

Comment:
I would be tempted to make this when verbosity is 2 and the rest when verbosity is 1
When I first ran the command, I ran without the `-v` flag and that it gives no feedback during the calculating part makes it tempting to kill the command

Reply:
This would require switching verbosity to be an int instead of a boolean, but I agree is a better approach. I will work in incorporating this.